src/main/java/net/fabricmc/loader/config/ConfigManagerImpl.java

```diff
@@ -0,0 +1,185 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.loader.config;
+
+import net.fabricmc.loader.api.FabricLoader;
+import net.fabricmc.loader.api.config.ConfigDefinition;
+import net.fabricmc.loader.api.config.serialization.ConfigSerializer;
+import net.fabricmc.loader.api.config.entrypoint.ConfigEnvironment;
+import net.fabricmc.loader.api.config.entrypoint.ConfigPostInitializer;
+import net.fabricmc.loader.api.config.data.DataCollector;
+import net.fabricmc.loader.api.config.data.DataType;
+import net.fabricmc.loader.api.config.exceptions.ConfigSerializationException;
+import net.fabricmc.loader.api.config.util.ListView;
+import net.fabricmc.loader.api.config.value.ValueKey;
+import net.fabricmc.loader.api.config.value.ValueContainer;
+import net.fabricmc.loader.api.config.entrypoint.ConfigInitializer;
+import net.fabricmc.loader.api.config.entrypoint.ConfigProvider;
+import net.fabricmc.loader.api.entrypoint.EntrypointContainer;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class ConfigManagerImpl {
+	public static final Logger LOGGER = LogManager.getLogger("Fabric|Config");
+	private static final Map<ConfigDefinition<?>, List<ValueKey<?>>> CONFIGS = new HashMap<>();
+	private static final Map<ConfigDefinition<?>, ListView<ValueKey<?>>> CONFIG_VIEWS = new HashMap<>();
+	private static final Map<String, ConfigDefinition<?>> CONFIG_DEFINITIONS = new ConcurrentHashMap<>();
+	private static final Map<String, ValueKey<?>> CONFIG_VALUES = new ConcurrentHashMap<>();
+
+	private static ListView<ConfigDefinition<?>> CONFIG_DEFINITION_VIEW = null;
+
+	private static boolean FINISHED = false;
+
+	public static boolean isFinished() {
+		return FINISHED;
+	}
+
+	public static ListView<ConfigDefinition<?>> getConfigKeys() {
+		return CONFIG_DEFINITION_VIEW;
+	}
+
+	public static ListView<ValueKey<?>> getValues(ConfigDefinition<?> configDefinition) {
+		return CONFIG_VIEWS.computeIfAbsent(configDefinition, definition ->
+				new ListView<>(CONFIGS.getOrDefault(definition, Collections.emptyList())));
+	}
+
+	public static @Nullable ValueKey<?> getValue(String configKeyString) {
+		return CONFIG_VALUES.get(configKeyString);
+	}
+
+	public static <R> @Nullable ConfigDefinition<R> getDefinition(String configKeyString) {
+		//noinspection unchecked
+		return (ConfigDefinition<R>) CONFIG_DEFINITIONS.get(configKeyString);
+	}
+
+	public static void initialize() {
+		if (FINISHED) return;
+
+		Map<String, Collection<ConfigInitializer<?>>> configInitializers = new HashMap<>();
+		Collection<ConfigPostInitializer> postInitializers = new ArrayList<>();
+
+		// We use one entrypoint to reduce the number of excess entrypoint keys
```

Comment:
lol what, entry points should be as specific as possible. That's why they have custom types. Your design is horrible and ignores the fact that entry point containers can be retrieved on demand!